src/distrs/beta.jl

```diff
@@ -22,3 +25,60 @@ function betalogpdf(α::T, β::T, x::T) where {T<:Real}
     val = xlogy(α - 1, y) + xlog1py(β - 1, -y) - logbeta(α, β)
     return x < 0 || x > 1 ? oftype(val, -Inf) : val
 end
+
+function betacdf(α::Real, β::Real, x::Real)
+    # Handle a degenerate case
+    if iszero(α)
+        return float(last(promote(α, β, x, x >= 0)))
```

Comment:
What's the motivation for this heuristic? And shouldn't the degenerate case maybe be handled in SpecialFunctions (IIRC you opened an issue)?

Reply:
When α is zero, the distribution is degenerate at zero so the CDF is just `x >= 0` but the return type would have to be adjusted.

I opened the issue in SpecialFunctions to discuss it there but I'm not completely sure that these edge cases belong in SpecialFunctions. Maybe they do but if you don't view the regularized incomplete beta function as a CDF then this limit might not be what you want. I assume the function might have applications outside of probability theory.

Reply:
Also in light of the discussion in SpecialFunctions, I agree that probably it's best to handle the special cases here but not in SpecialFunctions (as suggested also in https://github.com/JuliaMath/SpecialFunctions.jl/issues/378#issuecomment-1033369989).

Maybe the logic and the code would be slightly simpler here if we would focus on the special case `iszero(α) && !iszero(β) && iszero(x)` specifically (the other cases of `iszero(α)` should still be handled by SpecialFunctions, and in particular `iszero(α) && iszero(β)` should throw an error also here as it is not clear to which degenerate case it should correspond). I.e., one would only use
```julia
    # Not handled by SpecialFunctions but set to value of CDF of degenerate case here
    if iszero(α) && !iszero(β) && iszero(x)
        return float(one(first(promote(α, β, x))))
```

Reply:
There is also the question of the function value outside the unit interval which, again, is well defined for the CDF but not necessarily for the regularized incomplete beta function more generally so it might be simpler with the current version. The inequality definition is also the mathematical definition of CDF in the degenerate case so in some sense a very direct translation. Concretely, the branches based on the hypergeometric function also assume positive values of the parameters so I'd have to handle the zero cases outside there anyway.

Reply:
More generally, what I'm saying is that I don't see an issue with this branch covering a larger part than the parameter space than necessary. In particular because a single simple definition is sufficient. I realize, though, that I'll have to add a positivity condition to the non-zero parameter to avoid computing the wrong value in the case of `α = β = 0`.

I think it's fine to support the degenerate cases here even though we currently don't support them in Distributions. I suspect we might want to loosen the restriction in Distributions since it can be convenient e.g. during optimization. We loosened the requirement for the standard deviation in the Normal distribution a while back for the same reason.

Reply:
>  I realize, though, that I'll have to add a positivity condition to the non-zero parameter to avoid computing the wrong value in the case of α = β = 0.

Exactly, this was my main concern above 🙂

I'm fine with supporting the degenerate case, the condition in the `if` statement just seemed to cover too many other cases. But maybe it's simpler to only check `iszero(alpha) && !iszero(beta)` and cover all `x` instead of my suggestion above.

Reply:
I've updated the code.

Amusingly, I realized that
```
julia> StatsFuns.RFunctions.betacdf(0.0, 0.0, 0.0)
0.0

julia> StatsFuns.RFunctions.betacdf(0.0, 0.0, 1e-100)
0.5

julia> StatsFuns.RFunctions.betacdf(0.0, 0.0, 0.9999)
0.5

julia> StatsFuns.RFunctions.betacdf(0.0, 0.0, 1.0)
1.0
```
but I'm pretty sure that we shouldn't support the Bernoulli limit since the success probability depends on the ratio between `α` and `β` as they go to zero.

Reply:
Yes, I agree.